pkg/cmd/sync/sync_integration_test.go

```diff
@@ -0,0 +1,86 @@
+// +build integration
+
+package sync_test
+
+import (
+	"errors"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"github.com/jenkins-x/jx/pkg/cmd/clients"
+	"github.com/jenkins-x/jx/pkg/cmd/opts"
+	"github.com/jenkins-x/jx/pkg/cmd/sync"
+	"github.com/jenkins-x/jx/pkg/cmd/testhelpers"
+	"github.com/jenkins-x/jx/pkg/ksync"
+	"github.com/jenkins-x/jx/pkg/log"
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var sleepbeforecheck time.Duration = 30
+
+func TestKsyncCmd(t *testing.T) {
+	origJXHome, testJXHome, err := testhelpers.CreateTestJxHomeDir()
+	assert.NoError(t, err, "failed to create a test JX Home directory")
+
+	defer func() {
+		// Delete the pods in the test kubernetes cluster
+		kcli, err := ksync.NewCLI()
+		if err != nil {
+			log.Logger().Warnf("unable to create ksync cli client. Error: %s", err)
+		}
+
+		_, err = kcli.Clean()
+		if err != nil {
+			log.Logger().Warnf("unable to remove ksync daemonset. Error: %s", err)
+		}
+		log.Logger().Info("Removed ksync pods from cluster")
+
+		// Delete text directories
+		err = testhelpers.CleanupTestJxHomeDir(origJXHome, testJXHome)
+		if err != nil {
+			log.Logger().Warnf("unable to remove temporary directory %s: %s", testJXHome, err)
+		}
+		log.Logger().Info("Removed temp directory")
+	}()
+
+	commonOpts := opts.NewCommonOptionsWithTerm(clients.NewFactory(), os.Stdin, os.Stdout, os.Stderr)
+	assert.NoError(t, err, "error creating kubeclient")
+	cmd := sync.NewCmdSync(commonOpts)
+
+	executeCh := make(chan error, 1)
+	checkCh := make(chan error, 1)
+
+	// Run ksync in it's own goroutine and pass the error message into a channel
+	go func(cmd *cobra.Command) {
+		err := cmd.Execute()
+		executeCh <- err
+	}(cmd)
+
+	// Check for ksync daemonset in another go routine and pass the error to a channel
+	go func(opts *opts.CommonOptions) {
+		// Wait for at least one pod to be ready
+		time.Sleep(sleepbeforecheck * time.Second)
```

Comment:
I don't think this is going to fly. For the pipeline build, 30 seconds is probably a lot and we already have issues with a very slow test suite. On the other hand, I cannot get this test pass locally, because the sleep ends on my machine and internet connection before the download and install of ksync is complete. 

I think we need a more active readiness check. Either by explicitly watching the rollout of the daemon set deployment or by calling `kube.IsDaemonSetExists` in some sort of time-limited loop (maybe with exponential backoff). For the latter, we have several examples in the code.

Reply:
@ankitm123 what's about we skip this test for now and just add a test for the actual install of the binary. This would fix the imminent problem that the image is wrong. Once we have a better solution for writing this test we can then add it as additional test. WDYT?

Reply:
We can do that for the time being, but the issue was not with the binary (local ksync binary) being downloaded, the issue was with the tag for the ksync docker image. The binary was always getting installed and downloaded properly, but `jx sync` was using the same tag (0.4.4) for downloading docker images (kysnc images use the git hash, that is where the problem was). Adding the binary test will at least increase the test coverage for now, I will make a separate issue, and address the tests in a subsequent PR, what do u say?

Reply:
Sounds good. Thanks for the clarification.

Reply:
I made the changed requested, and added a simple check to see if ksync is installed or not. Once this PR is merged, I will work on adding the integration tests back with the changes requsted.